src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEMKernel.hpp

```diff
@@ -382,6 +383,396 @@ struct MassAndDampingMatrixKernel
+   * @param[in] xMax coordinate limits of the inner PML boundaries, right-back-bottom
+   * @param[in] dMin PML thickness, left-front-top
+   * @param[in] dMax PML thickness, right-back-bottom
+   * @param[in] cMin PML wave speed, left-front-top
+   * @param[in] cMax PML wave speed, right-back-bottom
+   * @param[in] r desired reflectivity of the PML
+   * @param[in] flagPML switch to flip between PML formulations
+   * @param[out] grad_n array holding the gradients at time n
+   * @param[out] divV_n array holding the divergence at time n
+   */
+  template< typename EXEC_POLICY, typename ATOMIC_POLICY >
+  void
+  launch( SortedArrayView< localIndex const > const targetSet,
+          arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const X,
+          traits::ViewTypeConst< CellElementSubRegion::NodeMapType > const elemToNodesViewConst,
+          arrayView1d< real64 const > const velocity,
+          arrayView1d< real64 const > const p_n,
+          arrayView2d< real64 const > const v_n,
+          arrayView1d< real64 const > const u_n,
+          real64 const (&xMin)[3],
+          real64 const (&xMax)[3],
+          real64 const (&dMin)[3],
+          real64 const (&dMax)[3],
+          real64 const (&cMin)[3],
+          real64 const (&cMax)[3],
+          real64 const r,
+          int const flagPML,
+          arrayView2d< real64 > const grad_n,
+          arrayView1d< real64 > const divV_n )
+  {
+
+    /// Loop over elements in the subregion, 'l' is the element index within the target set
+    forAll< EXEC_POLICY >( targetSet.size(), [=] GEOSX_HOST_DEVICE ( localIndex const l )
+    {
+      constexpr localIndex numNodesPerElem = FE_TYPE::numNodes;
+
+      /// global element index
+      localIndex const k = targetSet[l];
+
+      /// wave speed at the element
+      real64 const c = velocity[k];
+
+      /// coordinates of the element nodes
+      real64 xLocal[ numNodesPerElem ][ 3 ];
+
+      /// local arrays to store the pressure at all nodes and its gradient at a given node
+      real64 pressure[ numNodesPerElem ];
+      real64 pressureGrad[ 3 ];
+
+      /// local arrays to store the PML vectorial auxiliary variable at all nodes and its gradient at a given node
+      real64 auxV[3][ numNodesPerElem ];
+      real64 auxVGrad[3][3];
+
+      /// local arrays to store the PML scalar auxiliary variable at all nodes and its gradient at a given node
+      real64 auxU[ numNodesPerElem ];
+      real64 auxUGrad[3];
+
+      /// local array to store the PML damping profile
+      real64 sigma[ 3 ];
+
+      /// copy from global to local arrays
+      for( localIndex i=0; i<numNodesPerElem; ++i )
+      {
+        pressure[i] = p_n[elemToNodesViewConst[k][i]];
+        auxU[i] = u_n[elemToNodesViewConst[k][i]];
+        for( int j=0; j<3; ++j )
+        {
+          xLocal[i][j] =  X[elemToNodesViewConst[k][i]][j];
+          auxV[j][i] = v_n[elemToNodesViewConst[k][i]][j];
+        }
+      }
+
+      if( flagPML>1 )
```

Comment:
I forgot why you need the `flagPML` inside this `PMLKernel`? Isn't the kernel only launched inside `fsManager.apply< PerfectlyMatchedLayer >`?

Reply:
I kept it to switch between different PML implementations. Now I keep a single implementation so I removed this flag from the kernel.